src/plugins/dataflow-tool/microbit/communicator.md

```diff
@@ -0,0 +1,46 @@
+# Communicator
+
+The program for the micro:bit attached to the computer
+
+On received string via radio
+  - assemble the string
+  - write it to serial
+
+On received command string via serial
+  - broadcast it to targeted hub
+
+```js
+/**
+ * micro:bit + Dataflow: "Communicator"
+ * This is the basic program for the micro:bit physically attached to the computer.
+ */
+
+let mode = 0
+
+radio.setGroup(1)
+serial.redirect(
+    SerialPin.USB_TX,
+    SerialPin.USB_RX,
+    BaudRate.BaudRate9600
+)
+basic.showIcon(IconNames.Asleep)
+
+input.onButtonPressed(Button.A, function () {
+    mode = 1
+    basic.showIcon(IconNames.Happy)
+})
+
+input.onButtonPressed(Button.B, function () {
+    mode = 0
+    basic.showIcon(IconNames.Asleep)
+})
+
+radio.onReceivedString(function (receivedString) {
+    if (mode == 1) {
+        serial.writeLine(receivedString)
```

Comment:
If we run into issues with serial speed, You could skip sending the command messages to the computer. The communicator microbit is going to pick these up from other communicator microbits, but it doesn't really need to relay them to the computer.

Reply:
Oh good point.  I put an initial filter on.